reframe/frontend/loader.py

```diff
@@ -81,10 +81,9 @@ def _validate_check(self, check):
 
         name = type(check).__name__
         checkfile = os.path.relpath(inspect.getfile(type(check)))
-        required_attrs = ['sanity_patterns',
-                          'valid_systems', 'valid_prog_environs']
+        required_attrs = ['valid_systems', 'valid_prog_environs']
```

Comment:
I checked it and it works without this. The only problem I see generally from a user's point of view is that you get a stack trace for what is supposed to be a controlled error. This is not related to this PR, but it's something that we should think of.

Reply:
That's right. Removing this check will change the current behaviour and it will lead to a less user-friendly error.
I think we should keep this here - it makes total sense to me that the loader is the one that makes sure these 2 *special* variables are set.

Reply:
It's fine if we keep it for now, but I think we should somehow not raise an `AttributeError` in cases that a variable is required. It's not user-friendly and implies a programming error. If we make it some sort of `ReFrameError` we can then control the verbosity. I will open an issue specifically about this and merge this PR.

Reply:
I think there are two different issues here. One is using required variables in the framework side (as it is the case here), and the other is using these variables in a test.

Throwing an `AttributeError` from the framework is of course not user friendly, and as it is the case here, we catch that error and then we can either skip the test (as we do now) or raise a `ReframeError`. 
However, if a required variable is accessed during the execution of the test itself, that's a user's programming error, which makes it by all means an `AttributeError`. If this is the case, the user wants the error trace there so it knows where the error occurs.

Reply:
A user can still get the full trace if it were a `ReframeError` by increasing the verbosity, but getting a full stack trace for something like "variable 'x' is required", I don't like it as a user. I will open an issue and let's discuss it there. 

In this case, indeed, we still need this code segment here, because we skip the tests and not just fail. Although, if we fail, I'm also fine: less code and no "special" things.

Reply:
Issue #1899. Let the 🔥 begin 😄